onnxscript/ir/_core.py

```diff
@@ -1977,9 +1977,86 @@ def insert_before(self, node: Node, new_nodes: Iterable[Node] | Node, /) -> None
         self._nodes.insert_before(node, new_nodes)
 
     def sort(self) -> None:
```

Comment:
I would add a comment on what guarantees this method provides and add a breif docstring. Thanks! https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html